integration-tests/testutils/autopkgtest/autopkgtest_test.go

```diff
@@ -184,6 +186,28 @@ func (s *AutopkgtestSuite) TestAdtRunRemoteReturnsTplError(c *check.C) {
 	c.Assert(err, check.NotNil, check.Commentf("Expected error from tpl not received!"))
 }
 
+func (s *AutopkgtestSuite) TestAdtRunShellOnFail(c *check.C) {
+	scenarios := []struct {
+		shellOnFail     bool
+		testbedOptions  string
+		expectedOptions string
+	}{
+		{true, "testbed-options", "--shell-fail testbed-options"},
+		{false, "testbed-options", "testbed-options"},
+	}
+
+	for _, t := range scenarios {
+		s.subject.shellOnFail = t.shellOnFail
+		s.subject.adtRun(t.testbedOptions)
+
+		outputDir := outputDir(testArtifactsPath)
```

Comment:
It's generally best to not override a name like this. Makes logic unnecessarily context-dependent and slightly harder to read. Note, for example, that if that line is dropped altogether, the logic below likely still works, although incorrectly because outputDir is then a function.